html/modules/custom/reliefweb_job_tagger/src/Plugin/QueueWorker/OchaAiJobTagTaggerWorker.php

```diff
@@ -0,0 +1,255 @@
+
+    if (!$node || $node->bundle() !== 'job') {
+      \Drupal::logger('deubg')->notice('no bundle');
+      return;
+    }
+
+    if ($node->body->isEmpty()) {
+      \Drupal::logger('deubg')->notice('no body');
+      return;
+    }
+
+    // Only process it when fields are empty.
+    if (!$node->field_job_experience->isEmpty()) {
+      return;
+    }
+
+    if (!$node->field_career_categories->isEmpty()) {
+      return;
+    }
+
+    if (!$node->field_theme->isEmpty()) {
+      return;
+    }
+
+    // Load vocabularies.
+    $mapping = [];
+    $vocabularies = [
+      'job_experience',
+      'career_category',
+      'theme',
+    ];
+    foreach ($vocabularies as $vocabulary) {
+      $mapping[$vocabulary] = [];
+      $terms = $this->entityTypeManager->getStorage('taxonomy_term')->loadByProperties([
+        'status' => 1,
+        'vid' => $vocabulary,
+      ]);
+      /** @var \Drupal\taxonomy\Entity\Term $term */
+      foreach ($terms as $term) {
+        $mapping[$vocabulary][$term->getName()] = $term->getDescription() ?? $term->getName();
+      }
+    }
+
+    $data = $this->jobTagger
+      ->setVocabularies($mapping)
+      ->tag($node->getTitle(), $node->get('body')->value);
+
+    if (empty($data)) {
+      \Drupal::logger('deubg')->notice('no data');
+      return;
+    }
+
+    if (!isset($data['average']['mean_with_cutoff'])) {
+      \Drupal::logger('deubg')->notice('no average');
+      return;
+    }
+
+    // Use average mean with cutoff.
+    $data = $data['average']['mean_with_cutoff'];
+    $message = [];
+    $needs_save = FALSE;
+
+    if (isset($data['job_experience']) && $node->field_job_experience->isEmpty()) {
+      $term = $this->getRelevantTerm('job_experience', $data['job_experience'], 1);
+      $message[] = $this->setAiFeedback('Job experience', $data['job_experience'], [$term]);
+
+      $node->set('field_job_experience', $term);
+      $needs_save = TRUE;
+    }
+
+    if (isset($data['career_category']) && $node->field_career_categories->isEmpty()) {
+      $term = $this->getRelevantTerm('career_category', $data['career_category'], 1);
+      $message[] = $this->setAiFeedback('Career category', $data['career_category'], [$term]);
+
+      $node->set('field_career_categories', $term);
+      $needs_save = TRUE;
+    }
+
+    if (isset($data['theme']) && $node->field_theme->isEmpty()) {
+      $terms = $this->getRelevantTerm('theme', $data['theme'], 3);
+      $message[] = $this->setAiFeedback('Theme(s)', $data['theme'], $terms);
+
+      $node->set('field_theme', $terms);
+      $needs_save = TRUE;
+    }
+
+    if ($needs_save) {
+      if ($node->hasField('reliefweb_job_tagger_info')) {
+        $node->set('reliefweb_job_tagger_info', [
+          'value' => implode("\n\n", $message),
+          'format' => 'markdown',
+        ]);
+      }
+      $node->revision_log = 'Job has been updated by AI.';
+      $node->set('reliefweb_job_tagger_status', 'processed');
+      $node->isNewRevision();
```

Comment:
```suggestion
      $node->setNewRevision(TRUE);
```

I think this is meant to be a call to `setNewRevision`.

Reply:
True, fixed